### PR TITLE
data pod failure test suite changes for V9

### PR DESCRIPTION
**Pull request by sonalk0209**
# Problem Statement
Changes as per V9 for data pod failure test suite

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: tests/ha/single_pod_failure/test_data_pod_failure.py [hunk 1]
```diff
@@ -398,7 +417,6 @@ def test_writes_unsafe_pod_shutdown(self):
     # pylint: disable-msg=too-many-statements
     @pytest.mark.ha
     @pytest.mark.lc
-    @pytest.mark.skip(reason="Buckets cruds won't be supported with DTM0")
     @pytest.mark.tags("TEST-26444")
     @CTFailOn(error_handler)
```
Comment (rahul-hatwar) [inline tests/ha/single_pod_failure/test_data_pod_failure.py:421]: remove this at all places

File: tests/ha/single_pod_failure/test_data_pod_failure.py [hunk 2]
```diff
@@ -752,12 +775,11 @@ def test_pod_shutdown_delete_deployment(self):
...
     @pytest.mark.lc
-    @pytest.mark.skip(reason="Buckets cruds won't be supported with DTM0")
     @pytest.mark.tags("TEST-26445")
     @CTFailOn(error_handler)
     def test_deletes_during_pod_down(self):
         """
-        This test tests DELETEs during data pod down by deleting deployment
+        This test tests DELETEs during data pod down with replica method
         """
         LOGGER.info("STARTED: Test to verify DELETEs during data pod down by deleting deployment.")
```
Comment (rahul-hatwar) [inline tests/ha/single_pod_failure/test_data_pod_failure.py:784]: with replica method - check at all places once

File: tests/ha/single_pod_failure/test_data_pod_failure.py [hunk 3]
```diff
@@ -1954,10 +1984,10 @@ def test_mpu_during_pod_unsafe_shutdown(self):
     @pytest.mark.lc
     @pytest.mark.tags("TEST-32449")
     @CTFailOn(error_handler)
-    def test_partial_mpu_after_pod_unsafe_shutdown(self):
+    def test_partial_mpu_after_pod_shutdown(self):
         """
         This test tests degraded partial multipart upload after data pod unsafe shutdown
```
Comment (rahul-hatwar) [inline tests/ha/single_pod_failure/test_data_pod_failure.py:1989]: remove unsafe - look at other places too

File: tests/ha/single_pod_failure/test_data_pod_failure.py [hunk 4]
```diff
@@ -69,8 +67,6 @@ def setup_class(cls):
         Setup operations for the test file.
         """
         LOGGER.info("STARTED: Setup Module operations.")
-        cls.csm_user = CMN_CFG["csm"]["csm_admin_user"]["username"]
-        cls.csm_passwd = CMN_CFG["csm"]["csm_admin_user"]["password"]
         cls.num_nodes = len(CMN_CFG["nodes"])
         cls.username = []
```
Comment (rahul-hatwar) [inline tests/ha/single_pod_failure/test_data_pod_failure.py:71]: just make this as list(), check at all places

**Review by rahul-hatwar — APPROVED** [review]
Minor comment, LGTM

File: tests/ha/single_pod_failure/test_data_pod_failure.py [hunk 5]
```diff
@@ -646,18 +656,19 @@ def test_reads_during_pod_down(self):
...
         time.sleep(HA_CFG["common_params"]["30sec_delay"])
 
-        LOGGER.info("Step 3: Shutdown random data pod by deleting deployment and "
+        LOGGER.info("Step 3: Shutdown random data pod with replica method and "
                     "verify cluster & remaining pods status")
+        num_replica = self.num_replica - 1
         resp = self.ha_obj.delete_kpod_with_shutdown_methods(
             master_node_obj=self.node_master_list[0], health_obj=self.hlth_master_list[0],
-            down_method=const.RESTORE_DEPLOYMENT_K8S, event=event)
+            delete_pod=[self.delete_pod], num_replica=num_replica)
```
Comment (ApurwaMohite) [inline tests/ha/single_pod_failure/test_data_pod_failure.py:664]: We need to pass event to background IOs related tests.

File: tests/ha/single_pod_failure/test_data_pod_failure.py [hunk 6]
```diff
@@ -809,18 +818,19 @@ def test_deletes_during_pod_down(self):
...
         LOGGER.info("Step 3: Successfully started DELETEs in background")
 
-        LOGGER.info("Step 4: Shutdown random data pod by deleting deployment and "
+        LOGGER.info("Step 4: Shutdown random data pod with replica method and "
                     "verify cluster & remaining pods status")
+        num_replica = self.num_replica - 1
         resp = self.ha_obj.delete_kpod_with_shutdown_methods(
             master_node_obj=self.node_master_list[0], health_obj=self.hlth_master_list[0],
-            down_method=const.RESTORE_DEPLOYMENT_K8S, event=event)
+            delete_pod=[self.delete_pod], num_replica=num_replica)
```
Comment (ApurwaMohite) [inline tests/ha/single_pod_failure/test_data_pod_failure.py:826]: Same as above. Please check in all such tests.

File: tests/ha/single_pod_failure/test_data_pod_failure.py (hunk 4 above)
Comment (sonalk0209) [reply tests/ha/single_pod_failure/test_data_pod_failure.py:71]: Not seeing any codacy warning and not even getting any issues in test execution, so will keep it as is.

**Review by ankita-93 — APPROVED** [review]

**Review by ApurwaMohite — APPROVED** [review]

**Review by ravindrachoudhari — APPROVED** [review]

**Pull request merged**